apps/src/gamelab/GameLab.js

```diff
@@ -75,6 +81,40 @@ var MEDIA_PROXY = '//' + location.host + '/media?u=';
 // starts with http or https
 var ABSOLUTE_REGEXP = new RegExp('^https?://', 'i');
 
+var ASSET_PATH_PREFIX = "/v3/assets/";
+
+/**
+ * If the filename is relative (contains no slashes), then prepend
+ * the path to the assets directory for this project to the filename.
+ *
+ * If the filename URL is absolute, route it through the MEDIA_PROXY.
+ * @param {string} filename
+ * @returns {string}
+ */
+GameLab.maybeAddAssetPathPrefix = function (filename) {
+
+  if (ABSOLUTE_REGEXP.test(filename)) {
+    // We want to be able to handle the case where our filename contains a
+    // space, i.e. "www.example.com/images/foo bar.png", even though this is a
+    // technically invalid URL. encodeURIComponent will replace space with %20
+    // for us, but as soon as it's decoded, we again have an invalid URL. For
+    // this reason we first replace space with %20 ourselves, such that we now
+    // have a valid URL, and then call encodeURIComponent on the result.
+    return MEDIA_PROXY + encodeURIComponent(filename.replace(/ /g, '%20'));
+  }
+
+  filename = filename || '';
+  if (filename.length === 0) {
+    return '/blockly/media/1x1.gif';
+  }
+
+  if (filename.indexOf('/') !== -1 || !dashboard) {
+    return filename;
+  }
+
+  return ASSET_PATH_PREFIX + dashboard.project.getCurrentId() + '/' + filename;
+};
+
```

Comment:
I'd like to move this to studioApp at some point so it can be common. The `ASSET_PATH_PREFIX` is a little different in Applab as it has a mechanism to be overridden by test code. Also, I added the `!dashboard` check since I'm using `dashboard.project.getCurrentId()` instead of Applab's `Applab.channelId`